README.md

```diff
@@ -18,19 +18,24 @@ Starting with .NET 8, Source Link for the following source control providers is
 - [GitLab](https://gitlab.com) 12.0+ (for older versions see [GitLab settings](#gitlab))
 - [Bitbucket](https://bitbucket.org/) 4.7+ (for older versions see [Bitbucket settings](#bitbucket))
 
-If your project uses .NET SDK 8+ and is hosted by the above providers it does not need to reference any Source Link packages or set any build properties.
-
-**Otherwise**, you can enable Source Link experience in your project by setting a few properties and adding a PackageReference to a Source Link package specific to the provider:
+You can enable the Source Link experience in your project by opt-ing in to publish repository information:
```

Comment:
Perhaps we need to separate the properties from the PackageReferences in the project snippet.

Reply:
> I think the previous wording was better. `PublishRepositoryUrl` is not necessary to enable Source Link experience. It's only needed to publish URL to nupkg.

#1203 is evidence that users are confused by the current wording. Maybe the next step would be to clarify what "Source Link experience" users get without setting PublishRepositoryUrl, as it isn't clear to me either.

> Perhaps we need to separate the properties from the PackageReferences in the project snippet.

That was my intention with this change. Can you clarify what changes you'd like to see here, and I'll be happy to make them. Thanks!

Reply:
@tmat, I thought about this more and I think I came up with a better framing. I made a small addition to the opening section to capture that what we (I?) think of as "Source Link" is three components:

1. Embedding source information into the PDB
2. Adding source information into the .nuspec
3. Distributing PDBs

1 and 2 coupled in the current version of the docs, which is what lead to confusion in the linked bug. Currently, the docs make it look to some readers that in .NET 8+, 1 _and_ 2 happen automatically, but only 1 happens automatically. The previous version of this PR made it sound like 1 and 2 are both controlled by `<PublishRepositoryUrl>`, which isn't true either.

Now I think it's clearer that:

- 1 is controlled by including the package (or using a new SDK)
  - I also moved the `<EmbedUntrackedSources>` suggestion to this part as well, since it's coupled to the SDK decision and independent of the NuGet part
- .nuspec information is controlled by `<PublishRepositoryUrl>`

@tmat let me know what you think. Very happy to hear and include your feedback. Thanks!